File: JavaScript/Data Structure/Inverted Index/invertedIndex.js
class InvertedIndex {
    constructor() {
        this.index = {};
    }

    addDocument(doc) {
        const words = doc.Text.toLowerCase().split(' ');

        for (let i = 0; i < words.length; i++) {
            const chars = words[i];

            if (!this.index[chars]) {
                this.index[chars] = [];
            }
            if (!chars.includes(doc.ID)) {
                this.index[chars].push(doc.ID);
            }
        }
    }

    search(words) {
        return this.index.hasOwnProperty(words.toLowerCase()) ?
               this.index[words.toLowerCase()] : [];
    }
}

const invertedIndex = new InvertedIndex();

const docs = [
    { ID: 1, Text: "The quick brown fox" },
    { ID: 2, Text: "The quick blue fox" },
    { ID: 3, Text: "The fast red fox" }
];

for (let i = 0; i < docs.length; i++) {
    invertedIndex.addDocument(docs[i]);
}

console.log("Documents containing 'quick':", invertedIndex.search("quick"))
console.log("Documents containing 'fox':", invertedIndex.search("fox"))
console.log("Documents containing 'fast':", invertedIndex.search("fast"))
console.log("Documents containing 'elephant':", invertedIndex.search("elephant"))

File: Python/Data Structure/Inverted Index/inverted_index.py
class InvertedIndex:
    def __init__(self):
        self.index = {}
    
    def add_document(self, doc):
        words = doc['Text'].lower().split(" ")

        for char in words:
            if char not in self.index:
                self.index[char] = []
            if doc['ID'] not in self.index[char]:
                self.index[char].append(doc['ID'])
    
    def search(self, word):
        return self.index.get(word.lower(), [])

    

inverted_index = InvertedIndex()

docs = [
    {'ID': 1, 'Text': "The quick brown fox"},
	{'ID': 2, 'Text': "The quick blue fox"},
	{'ID': 3, 'Text': "The fast red fox"},
]

for item in docs:
    inverted_index.add_document(item)

print("Documents containing 'quick':", inverted_index.search("quick"))
print("Documents containing 'fox':", inverted_index.search("fox"))
print("Documents containing 'fast':", inverted_index.search("fast"))
print("Documents containing 'elephant':", inverted_index.search("elephant"))